src/main/java/io/aiven/kafka/connect/common/output/jsonwriter/KeyBuilder.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.output.jsonwriter;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+class KeyBuilder implements OutputFieldBuilder {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final JsonConverter converter;
+
+    public KeyBuilder() {
+        converter = new JsonConverter();
+        // TODO: make a more generic configuration
+        converter.configure(Map.of("schemas.enable", false, "converter.type", "value"));
+    }
+
+    /**
+     * Takes the {@link SinkRecord}'s key as a JSON.
+     *
+     * <p>If the key is {@code null}, it outputs nothing.
+     *
+     * <p>If the key is not {@code null}, it assumes the key <b>is</b> a JSON
+     *
+     * @param record       the record to get the key from
+     * @throws DataException when the key is not convertible to Json
+     */
+    @Override
+    public JsonNode build(final SinkRecord record) throws IOException {
+        Objects.requireNonNull(record, "record cannot be null");
+        Objects.requireNonNull(record.keySchema(), "key schema cannot be null");
+        if (record.key() == null) {
+            return null;
+        }
+
+        try {
+            return objectMapper.readTree(converter.fromConnectData(record.topic(), record.keySchema(), record.key()));
+        } catch (final IOException e) {
+            throw new DataException("failed");
+        }
```

Comment:
This exception, if thrown, will be pretty hard to debug as it hides the original exception. Same in few other similar places.

Also, the catch is a bit too broad for wrapping it into `DataException`. I'd probably narrow it to something like this:

```java
} catch (final SerializationException | JsonParseException e) {
    throw new DataException(e);
}
```

Reply:
I expected only a SerializationException here. Just because under the hood fromConnectData already transfrom it to a JsonNode. So...I would be very surprised if JsonParseException appears.

Reply:
Yep, makes sense. Then, `SerializationException` only

Reply:
- [ ] remove exceptions wrapper